api/tools/src/main/java/org/pmiops/workbench/tools/CreateTerraMethodSnapshot.java

```diff
@@ -0,0 +1,210 @@
+package org.pmiops.workbench.tools;
+
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.firecloud.ApiException;
+import org.pmiops.workbench.firecloud.model.FirecloudMethodID;
+import org.pmiops.workbench.firecloud.model.FirecloudMethodIO;
+import org.pmiops.workbench.firecloud.model.FirecloudMethodQuery;
+import org.pmiops.workbench.firecloud.model.FirecloudMethodResponse;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Create new Terra method snapshot in Agora
+ *
+ * <p>Assumptions - The provided Agora method namespace does not exist yet OR the cohort extraction
```

Comment:
I don't see code to create the namespace - does it just get lazily created when you create a method if it doesn't exist yet?